### PR TITLE
Feature(Engine) Resolve internal/external section references 

### PR DESCRIPTION
**Pull request by pereiramarco011**
**Proposed Changes**
- resolve same file section references
- resolve external file section references

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/resolver/file/file.go [hunk 1]
```diff
@@ -64,32 +66,49 @@ func (r *Resolver) Resolve(fileContent []byte, path string, resolveCount int, re
...
 
-func (r *Resolver) handleMap(value map[string]interface{}, path string,
+func (r *Resolver) handleMap(originalFileContent []byte, fullObject interface{}, value map[string]interface{}, path string,
 	resolveCount int, resolvedFilesCache map[string]ResolvedFile) (any, bool) {
 	for k, v := range value {
-		val, res := r.walk(v, path, resolveCount, resolvedFilesCache)
-		// check if it is a ref than everything needs to be changed
-		if res && strings.Contains(strings.ToLower(k), "ref") {
+		keyContainsRef := strings.Contains(strings.ToLower(k), "$ref")
+		isRef := keyContainsRef
```
Comment (cxMiguelSilva) [inline pkg/resolver/file/file.go:99]:
```suggestion
		isRef := strings.Contains(strings.ToLower(k), "$ref")
```

File: pkg/resolver/file/file.go [hunk 2]
```diff
@@ -142,135 +200,276 @@ func (r *Resolver) yamlWalk(value *yaml.Node, path string, resolveCount int, res
...
+		obj = fullObject
+	} else { // external file resolve
+		value = checkServerlessFileReference(value)
+
+		path := filepath.Join(filepath.Dir(filePath), value)
+		splitPath = strings.Split(path, "#") // splitting by removing the section to look for in the file
+		onlyFilePath := splitPath[0]
+		_, err := os.Stat(path)
+		if err != nil {
+			return *v, false
```
Comment (cxMiguelSilva) [inline pkg/resolver/file/file.go:230]:
```suggestion
                // if path is not valid or not exists return original value 
		onlyFilePath := splitPath[0]
		_, err := os.Stat(path)
		if err != nil {
			return *v, false
```

Add comments for the returns expressing what is returned and why

**Review by cxMiguelSilva — CHANGES_REQUESTED** [review]
Hi @pereiramarco011, 
Great work on these major changes in the KICS OpenAPI resolver.
 I added some suggestions to improve the code and the readability.

File: pkg/resolver/file/file.go (hunk 2 above)
Comment (pereiramarco011) [reply pkg/resolver/file/file.go:230]: Done here: https://github.com/Checkmarx/kics/pull/6405/commits/b36e81021f208c1e2ee9015f034e6cc69be9460b

**Review by cxMiguelSilva — APPROVED** [review]

**Review by gabriel-cx — APPROVED** [review]

File: test/main_test.go [hunk 3]
```diff
@@ -28,41 +28,41 @@ import (
...
-		"../assets/queries/azureResourceManager":            {FileKind: []model.FileKind{model.KindJSON}, Platform: "azureResourceManager"},
-		"../assets/queries/googleDeploymentManager/gcp":     {FileKind: []model.FileKind{model.KindYAML}, Platform: "googleDeploymentManager"},
-		"../assets/queries/googleDeploymentManager/gcp_bom": {FileKind: []model.FileKind{model.KindYAML}, Platform: "googleDeploymentManager"},
-		"../assets/queries/grpc":                            {FileKind: []model.FileKind{model.KindPROTO}, Platform: "grpc"},
-		"../assets/queries/buildah":                         {FileKind: []model.FileKind{model.KindBUILDAH}, Platform: "buildah"},
-		"../assets/queries/serverlessFW":                    {FileKind: []model.FileKind{model.KindYAML, model.KindYML}, Platform: "serverlessFW"},
-		"../assets/queries/knative":                         {FileKind: []model.FileKind{model.KindYAML}, Platform: "knative"},
+		//"../assets/queries/terraform/aws_bom":               {FileKind: []model.FileKind{model.KindTerraform, model.KindJSON}, Platform: "terraform"},
```
Comment (gabriel-cx) [inline test/main_test.go:31]: we should have all the tests uncommented by default

**Review by gabriel-cx — CHANGES_REQUESTED** [review]

File: test/main_test.go (hunk 3 above)
Comment (pereiramarco011) [reply test/main_test.go]: done :)

**Review by gabriel-cx — APPROVED** [review]

**Pull request merged**